solutions/src/main/java/ru/sberbank/school/task10/MyScalableConcurrentThreadPool.java

```diff
@@ -0,0 +1,217 @@
+    private boolean poolResize() throws InterruptedException {
+        int taskSize;
+        boolean needToInterrupt = false;
+        try {
+            tasksLock.lock();
+            taskSize = tasks.size();
+            try {
+                poolWriteLock.lock();
+                findFreeIndex();
+                int activeThreads = poolMap.size();
+                if (taskSize > activeThreads && taskSize > sizeMin && activeThreads < sizeMax) {
+
+                    if (!freeIndexForNewWorker.isEmpty()) {
+                        long id = freeIndexForNewWorker.take();
+
+                        ThreadWorker worker = new ThreadWorker("ThreadWorker " + id, id);
+                        poolMap.put(id, worker);
+                        worker.start();
+                    }
+                }
+                needToInterrupt = taskSize < activeThreads && activeThreads > sizeMin;
+            } finally {
+                poolWriteLock.unlock();
+            }
+        } finally {
+            tasksLock.unlock();
+        }
+        return needToInterrupt;
+    }
+
+    private void findFreeIndex() {
+        LongStream.range(0L, sizeMax)
+                .forEach(i -> {
+                    if (!poolMap.containsKey(i) && !freeIndexForNewWorker.contains(i)) {
+                        try {
+                            freeIndexForNewWorker.put(i);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                        }
+                    }
+                });
+    }
+
+    public int getActiveThreads() {
+        int count;
+        try {
+            poolReadLock.lock();
+            count = poolMap.size();
+        } finally {
+            poolReadLock.unlock();
+        }
+        return count;
+    }
+
+    private class ThreadWorker extends Thread {
+        private long id;
+
+        private ThreadWorker(String name, long id) {
+            super(name);
+            this.id = id;
+        }
+
+        @Override
+        public long getId() {
+            return id;
+        }
+
+        @Override
+        public void run() {
+            while (!isInterrupted()) {
+                try {
+                    boolean needToInterrupt = poolResize();
+
+                    if (needToInterrupt) {
+                        System.out.println("ThreadWorker " + id + " interrupted");
+                        poolWriteLock.lock();
+                        poolMap.remove(Thread.currentThread().getId());
+                        poolWriteLock.unlock();
+                        Thread.currentThread().interrupt();
+                        return;
+                    }
+
+                    FutureTask task = null;
+                    task = tasks.take();
+
+                    if (task != null) {
+                        task.run();
+                        System.out.println("ThreadWorker " + id + " ended");
+                    }
+                }catch (InterruptedException e) {
```

Comment:
WhitespaceAround: &#39;catch&#39; is not preceded with whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a312800406>*